tests/test_virus_scan_s3_bucket.py

```diff
@@ -187,18 +223,60 @@ def test_unfiltered(self, versions_page_size, dry_run, concurrency):
                 "already_tagged": 2,
             })
 
-    @pytest.mark.parametrize("concurrency", (0, 1, 3,))
-    @pytest.mark.parametrize("versions_page_size", (2, 4, 100,))
-    @pytest.mark.parametrize("dry_run", (False, True,))
-    def test_since_filtered(self, versions_page_size, dry_run, concurrency):
-        av_api_client, s3_client = self._get_mock_clients(self.versions_responses, versions_page_size)
+    def test_unfiltered_multi_bucket(self, versions_page_size, dry_run, concurrency):
+        av_api_client, s3_client = self._get_mock_clients(self.buckets_versions_responses, versions_page_size)
 
         with ThreadPoolExecutor(max_workers=concurrency) if concurrency else nullcontext() as executor:
             map_callable = map if executor is None else executor.map
             retval = virus_scan_bucket(
                 s3_client,
                 av_api_client,
-                "spade",
+                ("spade", "martello",),
+                prefix="",
+                since=None,
+                dry_run=dry_run,
+                map_callable=map_callable,
+            )
+
+        assert s3_client.mock_calls == [
+            mock.call.get_paginator("list_object_versions"),
+            mock.call.get_paginator().paginate(Bucket="spade", Prefix=""),
+            mock.call.get_paginator("list_object_versions"),
+            mock.call.get_paginator().paginate(Bucket="martello", Prefix=""),
+        ]
```

Comment:
Do these calls need to be sorted for threading as well?

Reply:
_Shouldn't_ - the iterator is only ever advanced by the main thread.

Reply:
Ah yes - that's called before the map_callable is applied, isn't it. Cool.